### PR TITLE
[libc][stdfix] Add support for fixed point types in the testing infrastructure.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

File: libc/test/UnitTest/LibcTest.cpp [hunk 1]
```diff
@@ -53,6 +55,17 @@ describeValue(ValType Value) {
   return cpp::to_string(Value);
 }
 
+#ifdef LIBC_COMPILER_HAS_FIXED_POINT
+template <typename T>
+cpp::enable_if_t<cpp::is_fixed_point_v<T>, cpp::string> describeValue(T Value) {
+  using FXRep = fixed_point::FXRep<T>;
+  using comp_t = typename FXRep::CompType;
+
+  return cpp::to_string(cpp::bit_cast<comp_t>(Value)) + " * 2^-" +
```
Comment (michaelrj-google) [inline libc/test/UnitTest/LibcTest.cpp:64]: could you add a TODO to switch this to proper fixed point printing when that's available?

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Review by nickdesaulniers — APPROVED** [review]

**Review by PiJoules — APPROVED** [review]

File: libc/test/UnitTest/LibcTest.cpp [hunk 2]
```diff
@@ -8,9 +8,11 @@
 
 #include "LibcTest.h"
 
+#include "include/llvm-libc-macros/stdfix-macros.h"
```
Comment (gchatelet) [inline libc/test/UnitTest/LibcTest.cpp:11]: This broke the Bazel build.

File: libc/src/__support/fixed_point/fx_rep.h [hunk 3]
```diff
@@ -10,144 +10,245 @@
...
+
+template <int Bits> struct Storage {
+  static_assert(Bits > 0 && Bits <= 64, "Bits has to be between 1 and 64.");
+  using Type = typename cpp::conditional_t<
+      (Bits <= 8), uint8_t,
+      typename cpp::conditional_t<
+          (Bits <= 16 && Bits > 8), uint16_t,
+          typename cpp::conditional_t<(Bits <= 32 && Bits > 16), uint32_t,
+                                      uint64_t>>>;
+};
```
Comment (gchatelet) [inline libc/src/__support/fixed_point/fx_rep.h:32]: In C++20 we can express this logic with regular functions :)
https://godbolt.org/z/cn1qqTWez

Comment (nickdesaulniers) [reply libc/src/__support/fixed_point/fx_rep.h:32]: This is blowing my mind right now. :exploding_head:

Comment (lntue) [reply libc/src/__support/fixed_point/fx_rep.h:32]: That looks much better than these types of conditional chains.

Comment (lntue) [reply libc/src/__support/fixed_point/fx_rep.h:32]: We might have to replicate this logic in C, probably inside `include/llvm-libc-types/stdfix_types.h` in order to add definitions for `int_*_t` and `uint_*_t` as specified in the standard (section 7.18a.2).  Then probably we can remove this conditional template and use those instead.

**Pull request merged**